### DISCUSSION TITLE
Adding id attribute to text or headings

### DISCUSSION THREAD

**Discussion by thomascoppein** [craftcms/ckeditor#174] [Q&A] [answered]
### Description

Is there a way to add an id to text or heading so it can be targeted by anchor link?

**Comment by i-just** [comment, marked as answer]
Hi, thanks for reaching out! You could use the [anchors plugin](https://github.com/craftcms/anchors/) to add anchor links to selected heading levels.

Adding custom IDs or other attributes directly in CKEditor is not currently supported: https://github.com/ckeditor/ckeditor5/issues/7369, but someone has created a plugin that adds the ability to add an id attribute: https://github.com/rhysstubbs/ckeditor5-add-attribute-to-element. You could use that as a starting point to create your own or pull in this one yourself.

**Comment by sweetroll** [comment]
This is now supported via the Bookmark button for anyone needing this

**Reply by thisisjamessmith** [reply]
It seems that the bookmark button will simply add empty anchor tags instead of more cleanly adding an `id` if there's a dedicated element selected such as a heading (as will typically be the case I'd imagine most of the time).

<img width="183" height="64" alt="image" src="https://github.com/user-attachments/assets/8a304851-fbcb-4482-81fa-c818b727f338" />

Would be good to see this cleaned up.
